homeassistant/components/madvr/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""Coordinator for handling data fetching and updates."""
+
+import logging
+from typing import Any
+
+from madvr.madvr import Madvr
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
```

Comment:
Idem
You might need a `if TYPE_CHECKING:` block

Reply:
I had this before but then you said to remove it. If I use if TYPE_CHECKING here, I have to use it below which you wanted me to remove

Reply:
ooooooh

```py
if TYPE_CHECKING:
    from . import MadVRConfigEntry
```
You also probably have to add `from __future__ import annotations`